proxy/src/proxy.rs

```diff
@@ -236,6 +245,8 @@ struct Client<'a, S> {
     params: &'a StartupMessageParams,
     /// Unique connection ID.
     session_id: uuid::Uuid,
+    /// Varous caches for the cloud API responses.
+    caches: &'static auth::caches::ApiCaches,
```

Comment:
Seems weird for caches to be a property of a connection. It's shared by all connections.